### PR TITLE
4.14 GA Cloud Compute Bug Batching 2

### PR DESCRIPTION
**Pull request by cbippley**
Version(s):
4.14

Issue:
[Bug Batching for 4.14 GA](https://issues.redhat.com/browse/OSDOCS-5987)

Link to docs preview:
[Preview](https://66843--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-14-release-notes#ocp-4-14-bug-fixes)

QE review:
- [ ] QE has approved this change.
not needed

Additional information:
[Cloud compute bug filter](https://issues.redhat.com/secure/Dashboard.jspa?selectPageId=12347491#SIGwKWmOqDAaIiGh1DSCAiYg6oaN4RY8HF6AMAwlQMHEVU1ccdWpPFMCNc1SXIGgGCNeaPAtG0HT+ogCErGsyGEN+BWVIiiIKPs54OktK2uutTXLQw21YBte3QawILgh+5JftSCgaB47qFfMcT8UaPCIpglTgdQGgaM1dQLIU7oxHCt27BaQlWEAA)

### CODE REVIEW

File: release_notes/ocp-4-14-release-notes.adoc [hunk 1]
```diff
@@ -1678,10 +1678,30 @@ Targeting {product-title} 4.15, the OpenShift SDN CNI network plugin will not be
 
 * Previously, after the rotation of cloud credentials in the `clouds.yaml` file, the OpenStack machine API provider would need to be restarted in order to pick up the new cloud credentials. As a result, the ability of a machine set to scale to zero could be affected. With this change, cloud credentials are no longer cached, and the provider reads the corresponding secret freshly as needed. (link:https://issues.redhat.com/browse/OCPBUGS-8687[*OCPBUGS-8687*])
 
- * Previously, some conditions during the startup process of the Cluster Autoscaler Operator caused a lock that prevented the Operator from successfully starting and marking itself available. As a result, the cluster became degraded. The issue is resolved with this release. (link:https://issues.redhat.com/browse/OCPBUGS-20038[*OCPBUGS-20038*])
+* Previously, some conditions during the startup process of the Cluster Autoscaler Operator caused a lock that prevented the Operator from successfully starting and marking itself available. As a result, the cluster became degraded. The issue is resolved with this release. (link:https://issues.redhat.com/browse/OCPBUGS-20038[*OCPBUGS-20038*])
```
Comment (bburt-rh) [inline release_notes/ocp-4-14-release-notes.adoc:1681]:
```suggestion
* Previously, some conditions during the startup process of the Cluster Autoscaler Operator caused a lock that prevented the Operator from successfully starting and marking itself as available. As a result, the cluster became degraded. The issue is resolved with this release. (link:https://issues.redhat.com/browse/OCPBUGS-20038[*OCPBUGS-20038*])
```
- Nit: Adding `as` sounds more idiomatic to me. WDYT?

File: release_notes/ocp-4-14-release-notes.adoc [hunk 2]
```diff
@@ -1678,10 +1678,30 @@ Targeting {product-title} 4.15, the OpenShift SDN CNI network plugin will not be
...
+
+* Previously, the Control Plane Machine Set Operator used the `Infrastructure` object specification to determine the platform type for the cluster. For clusters upgraded from {product-title} version 4.5 and earlier, this meant that the Operator could not correctly determine that a cluster was running on AWS, and therefore did not generate the `ControlPlaneMachineSet` custom resource (CR) as expected. With this release, the Operator uses the status platform type, which is populated on all clusters independent of when they were created, and is now able to generate the `ControlPlaneMachineSet` CR for all clusters. (link:https://issues.redhat.com/browse/OCPBUGS-11389[*OCPBUGS-11389*])
```
Comment (bburt-rh) [inline release_notes/ocp-4-14-release-notes.adoc:1689]:
```suggestion
* Previously, the Control Plane Machine Set Operator used the `Infrastructure` object specification to determine the platform type for the cluster. For clusters upgraded from {product-title} version 4.5 and earlier, this [what?... practice? situation?] meant that the Operator could not correctly determine that a cluster was running on AWS, and therefore did not generate the `ControlPlaneMachineSet` custom resource (CR) as expected. With this release, the Operator uses the status platform type, which is populated on all clusters independent of when they were created and is now able to generate the `ControlPlaneMachineSet` CR for all clusters. (link:https://issues.redhat.com/browse/OCPBUGS-11389[*OCPBUGS-11389*])
```
- Per [IBM Style](https://www.ibm.com/docs/en/ibm-style?topic=grammar-pronouns#ambiguous-pronoun-references), ensure that pronouns have clear references.
- Remove comma before subordinating conjunction `and`. A comma is only required before a coordinating conjunction.

File: release_notes/ocp-4-14-release-notes.adoc [hunk 3]
```diff
@@ -1678,10 +1678,30 @@ Targeting {product-title} 4.15, the OpenShift SDN CNI network plugin will not be
...
+
+* Previously, machines created by a control plane machine set were considered ready once the underlying Machine API machine was running. With this release, the machine is not considered ready until the node linked to that machine is also ready. (link:https://issues.redhat.com/browse/OCPBUGS-7989[*OCPBUGS-7989*])
+
+* Previously, the Control Plane Machine Set Operator prioritized failure domains alphabetically and moved machines from alphabetically later failure domains to alphabetically earlier failure domains, even if doing so did not improve the the availability of the machines across the failure domains. With this release, the Operator is updated to prioritize failure domains that are present in the existing machines and respect existing failure domains that provide better availability. (link:https://issues.redhat.com/browse/OCPBUGS-7921[*OCPBUGS-7921*])
```
Comment (bburt-rh) [inline release_notes/ocp-4-14-release-notes.adoc:1693]:
```suggestion
* Previously, the Control Plane Machine Set Operator prioritized failure domains alphabetically and moved machines from alphabetically later failure domains to alphabetically earlier failure domains, even if doing so did not improve the the availability of the machines across the failure domains. With this release, the Operator is updated to prioritize failure domains that are present in the existing machines and to respect existing failure domains that provide better availability. (link:https://issues.redhat.com/browse/OCPBUGS-7921[*OCPBUGS-7921*])
```
- add `to` to make grammatically parallel with `to prioritize`

File: release_notes/ocp-4-14-release-notes.adoc [hunk 4]
```diff
@@ -1678,10 +1678,30 @@ Targeting {product-title} 4.15, the OpenShift SDN CNI network plugin will not be
...
+
+* Previously, when creating an {product-title} cluster on Nutanix that uses DHCP instead of an IP address management (IPAM) network configuration, the hostname of the VM was not set by DHCP. With this release, the VM hostname is set with values from the ignition configuration files. As a result, the issue is resolved for DHCP as well as other network configuration types. (link:https://issues.redhat.com/browse/OCPBUGS-6727[*OCPBUGS-6727*])
+
+* Previously, multiple clusters could be created in the `openshift-cluster-api` namespace. This namespace should contain only one cluster. With this release, additional clusters cannot be created in this namespace. (link:https://issues.redhat.com/browse/OCPBUGS-4147[*OCPBUGS-4147*])
```
Comment (bburt-rh) [inline release_notes/ocp-4-14-release-notes.adoc:1701]:
```suggestion
* Previously, multiple clusters could be created in the `openshift-cluster-api` namespace. This namespace must contain only one cluster. With this release, additional clusters cannot be created in this namespace. (link:https://issues.redhat.com/browse/OCPBUGS-4147[*OCPBUGS-4147*])
```
Per IBM Style, avoid `should` and don't use it as a synonym for `must`. The word `should` can have an ambiguous meaning.

File: release_notes/ocp-4-14-release-notes.adoc [hunk 5]
```diff
@@ -1678,10 +1678,30 @@ Targeting {product-title} 4.15, the OpenShift SDN CNI network plugin will not be
...
+
+* Previously, clearing some parameters from the `providerSpec` field of a control plane machine set custom resource caused a loop of control plane machine deletion and creation. With this release, these parameters receive a default value if they are cleared or left empty. (link:https://issues.redhat.com/browse/OCPBUGS-2960[*OCPBUGS-2960*])
```
Comment (bburt-rh) [inline release_notes/ocp-4-14-release-notes.adoc:1703]:
```suggestion
* Previously, clearing some parameters from the `providerSpec` field of a control plane machine set custom resource caused a loop of control plane machine deletion and creation. With this release, these parameters receive a default value if they are cleared or left empty, which resolves the issue. (link:https://issues.redhat.com/browse/OCPBUGS-2960[*OCPBUGS-2960*])
```

**Review by bburt-rh — COMMENTED** [review]
Excellent PR! Very solid. I've added a few nitpicky comments and suggestions for you to consider. Lmk if you have any questions. Thanks!

**Comment by bscott-rh** [comment]
Thanks Joe. Merge review LGTM. Merged to enterprise-4.14.

**Pull request merged**